wandb/sdk/launch/builder/build.py

```diff
@@ -499,3 +509,75 @@ def _create_docker_build_ctx(
 def join(split_command: List[str]) -> str:
     """Return a shell-escaped string from *split_command*."""
     return " ".join(shlex.quote(arg) for arg in split_command)
+
+
+def make_image_uri(
+    builder: AbstractBuilder,
+    launch_project: LaunchProject,
+    repository: Optional[Any],
+    entry_point: EntryPoint,
+    docker_args: dict,
+) -> Optional[str]:
+    """
+    Helper for testing
+    """
+    image_uri: Optional[str] = builder.build_image(
+        launch_project,
+        repository,
+        entry_point,
+        docker_args,
+    )
+    return image_uri
+
+
+def build_image_from_project(
+    launch_project: LaunchProject,
+    api: Api,
+    build_type: Optional[str],
+    launch_config: Optional[Dict] = None,
+) -> str:
+    """
+    Accepts a reference to the Api class and a pre-computed launch_spec
+    object, with an optional launch_config to set git-things like repository
+    which is used in naming the output docker image, and build_type defaulting
+    to docker (but could be used to build kube resource jobs w/ "kaniko")
+
+    updates launch_project with the newly created docker image uri and
+    returns the uri
+    """
+    assert launch_project.uri, "To build an image on queue a URI must be set."
+
+    builder_config = {"type": build_type}
+    docker_args = {}
+    repository = None
+    if launch_config:
+        repository = launch_config.get("url")
+
+    if launch_project.python_version:
+        docker_args["python_version"] = launch_project.python_version
+
+    if launch_project.cuda_version:
+        docker_args["cuda_version"] = launch_project.cuda_version
+
+    if launch_project.docker_user_id:
+        docker_args["user"] = str(launch_project.docker_user_id)
+
+    wandb.termlog(f"{LOG_PREFIX}Building docker image from uri source.")
+    launch_project = fetch_and_validate_project(launch_project, api)
+    builder = load_builder(builder_config)
+    entry_point: EntryPoint = launch_project.get_single_entry_point() or EntryPoint(
+        name="main.py",
```

Comment:
This should be `EntrypointDefaults.PYTHON[-1]`

Reply:
I did not address this because I'm not sure I exactly know what you mean.

Reply:
since `EntrypointDefaults` is a list = `["python", "main.py"]` we should just use the last value of the array instead of hard coding `main.py` so it changes with the default entrypoint.